modin/core/dataframe/algebra/groupby.py

```diff
@@ -289,31 +289,37 @@ def caller(
         The same type as `query_compiler`
             QueryCompiler which carries the result of GroupBy aggregation.
         """
+        is_unsupported_axis = axis != 0
+        is_empty_data = (
+            len(query_compiler.columns) == 0 or len(query_compiler.index) == 0
+        )
+        is_grouping_on_external_grouper = (
+            groupby_kwargs.get("level", None) is None and by is not None
+        )
+        is_unsupported_external_grouper = isinstance(by, pandas.Grouper) or (
+            not hashable(by) and not isinstance(by, type(query_compiler))
+        )
```

Comment:
We should probably add a comment on what we treat as an external grouper.

Reply:
Renamed the variables as `is_grouping_using_by_arg` and `is_unsupported_by_arg` to get rid of this confusion